cmake/toolchain.cmake

```diff
@@ -52,6 +53,23 @@ set(CMAKE_CXX_COMPILER_WORKS TRUE)
 # Project setup
 enable_language(C CXX ASM)
 
+# Set the language standard to use per target
+function(mbed_set_language_standard target)
+    set_target_properties(${target}
+        PROPERTIES
+            C_STANDARD 11
+            C_STANDARD_REQUIRED YES
+            C_EXTENSIONS NO
+    )
+
+    set_target_properties(${target}
+        PROPERTIES
+            CXX_STANDARD 14
+            CXX_STANDARD_REQUIRED YES
+            CXX_EXTENSIONS NO
```

Comment:
Please note that because `CXX_EXTENSIONS` is set to `NO`, we will get `-std=c++14` instead of `-std=gnu++14` (similar difference for the C standard).

Are we Ok with that?

Reply:
why is it set to NO ? I recall we got a code requiring gnu extensions (that was definitely the case for ARMC5).

Reply:
it would be nice to keep the gnu extensions.

Reply:
[Done](https://github.com/ARMmbed/mbed-os/compare/82076cf967c6bd11a7336153a8d55b05a849ec1f..a8e482b680a37799737c419c9225c4731b58eff0).
It was not building without the gnu extension.